examples/tutorials/streamwriter_basic_tutorial.py

```diff
@@ -0,0 +1,658 @@
+"""
+StreamWriter Basic Usage
+========================
+
+This tutorial shows how to use :py:class:`torchaudio.io.StreamWriter` to
+encode and save audio/video data into various formats/destinations.
+
+"""
+
+######################################################################
+#
+# .. note::
+#
+#    This tutorial requires torchaudio nightly build and FFmpeg libraries (>=4.1, <4.4).
+#
+#    To install torchaudio nightly build, please refer to
+#    https://pytorch.org/get-started/locally/ .
+#
+#    There are multiple ways to install FFmpeg libraries.
+#    If you are using Anaconda Python distribution,
+#    ``conda install 'ffmpeg<4.4'`` will install the required FFmpeg libraries.
+#
+
+######################################################################
+#
+# .. warning::
+#
+#    TorchAudio dynamically loads compatible FFmpeg libraries
+#    installed on the system.
+#    The types of supported formats (media format, encoder, encoder
+#    options etc) depend on the libraries.
+#
+#    To check the available muxers and encoders, you can use the
+#    following command
+#
+#    .. code-block:: console
+#
+#       ffmpeg -muxers
+#       ffmpeg -encoders
+
+######################################################################
+#
+# Preparation
+# -----------
+
+import torch
+import torchaudio
+
+print(torch.__version__)
+print(torchaudio.__version__)
+
+######################################################################
+#
+
+try:
+    from torchaudio.io import StreamWriter
+except ImportError:
+    try:
+        import google.colab
+
+        print(
+            """
+            To enable running this notebook in Google Colab, install nightly
+            torch and torchaudio builds by adding the following code block to the top
+            of the notebook before running it:
+            !pip3 uninstall -y torch torchvision torchaudio
+            !pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu
+            """
+        )
+    except ModuleNotFoundError:
+        pass
+    raise
+
+print("FFmpeg library versions")
+for k, v in torchaudio.utils.ffmpeg_utils.get_versions().items():
+    print(f"  {k}: {v}")
+
+######################################################################
+#
+
+import io
+import os
+import tempfile
+
+from torchaudio.utils import download_asset
+from IPython.display import Audio, Video
+
+SAMPLE_PATH = download_asset("tutorial-assets/Lab41-SRI-VOiCES-src-sp0307-ch127535-sg0042.wav")
+WAVEFORM, SAMPLE_RATE = torchaudio.load(SAMPLE_PATH, channels_first=False)
```

Comment:
shall we mention the shape of the encoded waveform must be of shape `(time, channel)`?

Reply:
It's explained later. I want to keep preparation section minimum and less important so that it's okay to scroll the entire section.